pkg/identity/cache/allocation_test.go

```diff
@@ -100,17 +104,120 @@ func (e *IdentityAllocatorConsulSuite) SetUpTest(c *C) {
 	kvstore.SetupDummy("consul")
 }
 
-type dummyOwner struct{}
+type dummyOwner struct{
+	updated   chan identity.NumericIdentity
+	mutex     lock.Mutex
+	cache     IdentityCache
+}
 
-func (d dummyOwner) TriggerPolicyUpdates(force bool, reason string) {
+func newDummyOwner() *dummyOwner {
+	return &dummyOwner{
+		cache:   IdentityCache{},
+		updated: make(chan identity.NumericIdentity, 1024),
+	}
 }
 
-func (d dummyOwner) GetNodeSuffix() string {
+func (d *dummyOwner) UpdateIdentities(added, deleted IdentityCache) {
+	d.mutex.Lock()
+	log.Debugf("Dummy UpdateIdentities(added: %v, deleted: %v)", added, deleted)
+	for id, lbls := range added {
+		d.cache[id] = lbls
+		d.updated <- id
+	}
+	for id, _ := range deleted {
+		delete(d.cache, id)
+		d.updated <- id
+	}
+	d.mutex.Unlock()
+}
+
+func (d *dummyOwner) GetIdentity(id identity.NumericIdentity) labels.LabelArray {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+	return d.cache[id]
+}
+
+func (d *dummyOwner) GetNodeSuffix() string {
 	return "foo"
 }
 
+// WaitUntilCount waits until the cached identity count reaches
+// 'target' and returns the number of events processed to get
+// there. Returns 0 in case of 'target' is not reached in 60 seconds.
+func (d *dummyOwner) WaitUntilID(target identity.NumericIdentity) int {
+	rounds := 0
+	for {
+		select {
+		case nid, ok := <-d.updated:
+			if (!ok) {
+				// updates channel closed
+				return 0
+			}
+			rounds++
+			if nid == target {
+				return rounds
+			}
+		case <-time.After(60 * time.Second):
```

Comment:
This is a testing utility, so the exact timeout semantics is not very significant. I kept the function as is and changed the function comment to end in:
```
// Returns 0 in case of 'd.updated' channel is closed or
// nothing is received from that channel in 60 seconds.
```